pipeline.py

```diff
@@ -69,31 +93,123 @@ class TextProcessingPipeline:
+                cleaned_tokens.append(processed_token.get_cleaned())
+                single_tagged_tokens.append(processed_token.get_single_tagged())
+                multiple_tagged_tokens.append(processed_token.get_multiple_tagged())
+
+            article.save_as(' '.join(cleaned_tokens), 'cleaned')
+            article.save_as(' '.join(single_tagged_tokens), 'single_tagged')
+            article.save_as(' '.join(multiple_tagged_tokens), 'multiple_tagged')
 
     def _process(self, raw_text: str):
         """
         Processes each token and creates MorphToken class instance
         """
-        pass
+        # Gets rid of '-\n's that break up words, so we won't get 'кот-' 'орый' as example (Should be: 'который')
+        text = raw_text.replace('-\n', '')
+
+        # [:500] is there because scientific articles are huge and CI takes too long instead (way more than 5 mins)
+        result = Mystem().analyze(text[:500])
+
+        morphological_tokens = []
+
+        for token_info in result:
+            original_word = token_info['text']
+
+            if not re.match(r'[а-яА-Яa-zA-Z]', original_word):
+                continue
+
+            morphological_token = MorphologicalToken(original_word=original_word)
+
+            if 'analysis' not in token_info:
+                continue
+
+            if not token_info['analysis']:
+                continue
+
+            if 'lex' not in token_info['analysis'][0] or 'gr' not in token_info['analysis'][0]:
+                continue
+
+            morphological_token.normalized_form = token_info['analysis'][0]['lex']
+            morphological_token.tags_mystem = token_info['analysis'][0]['gr']
+
+            morph = pymorphy2.MorphAnalyzer()
+            parsed_word = morph.parse(original_word)[0]
+            morphological_token.tags_pymorphy = parsed_word.tag
+
+            morphological_tokens.append(morphological_token)
+
+        return morphological_tokens
 
 
 def validate_dataset(path_to_validate):
     """
     Validates folder with assets
     """
-    pass
+
+    pathlib_path_to_validate = pathlib.Path(path_to_validate)
+
+    if not path_to_validate.exists():
+        raise FileNotFoundError
+
+    if not pathlib_path_to_validate.is_dir():
+        raise NotADirectoryError
+
+    file_ids = []
+
+    for full_file_name in pathlib_path_to_validate.iterdir():
+        file_name = full_file_name.name
+
+        # In case this annoying macOS Finder system file exists locally
+        if file_name == '.DS_Store':
+            full_file_name.unlink()
+            continue
+
+        match = re.match(r'\d+', file_name)
+
+        if not match:
+            raise InconsistentDatasetError
+
+        file_id = int(match.group(0))
+
+        file_ids.append(file_id)
+
+    file_ids = sorted(file_ids)
+
+    last_file_id = 0
+
+    for file_id in file_ids:
+        if not last_file_id and file_id != 1 or not 0 <= file_id - last_file_id <= 1:
```

Comment:
the check is very strange and also there is an extremely strange condition - how do you expect the difference to be negative? why do you expect numbers to be in a range from 0 to 1 inclusive?